lib/ts/recipe/emailpassword/components/themes/default/emailVerification/emailVerificationScreen/sendVerifyEmail.tsx

```diff
@@ -0,0 +1,134 @@
+/* Copyright (c) 2020, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Imports.
+ */
+import React, { PureComponent } from "react";
+import StyleContext from "../../../../styles/styleContext";
+
+import { SendVerifyEmailThemeProps, SendVerifyEmailThemeState } from "../../../../../types";
+
+/** @jsx jsx */
+import { jsx } from "@emotion/react";
+import EmailIcon from "../../../../../assets/emailLargeIcon";
+import ArrowRightIcon from "../../../../../assets/arrowRightIcon";
+import { API_RESPONSE_STATUS, SEND_VERIFY_EMAIL_STATUS } from "../../../../../constants";
+import { SOMETHING_WENT_WRONG_ERROR } from "../../../../../../../constants";
+
+/*
+ * Component.
+ */
+
+export default class SendVerifyEmail extends PureComponent<SendVerifyEmailThemeProps, SendVerifyEmailThemeState> {
+    static contextType = StyleContext;
+    /*
+     * Constructor.
+     */
+    constructor(props: SendVerifyEmailThemeProps) {
+        super(props);
+        this.state = {
+            status: SEND_VERIFY_EMAIL_STATUS.READY
+        };
+    }
+
+    /*
+     * Methods.
+     */
+
+    onSuccess = (): void => {
+        this.setState(() => ({
+            status: SEND_VERIFY_EMAIL_STATUS.SUCCESS
+        }));
+        if (this.props.onSuccess !== undefined) {
+            this.props.onSuccess();
+        }
+    };
+
+    sendEmail = async (): Promise<void> => {
+        const { callAPI } = this.props;
+        try {
+            const response = await callAPI();
+
+            if (response.status === API_RESPONSE_STATUS.OK) {
```

Comment:
Another API response is that the email has already been verified. That's not being handled.

Reply:
Good catch. This should be really edge case as we first check in the feature if email is already verified before loading this page. https://github.com/supertokens/supertokens-auth-react/pull/115/files#diff-95106643fb53b192624f3131a1ca70f9276079cf6e9dd20517d11445e89d51d2R224

But if someone comes back to this page without reloading after they have validated their email, then we should simply redirect to onSuccessRedirect.